definitions/checks/foreman/check_puppet_capsules.rb

```diff
@@ -0,0 +1,38 @@
+module Checks
+  class CheckPuppetCapsules < ForemanMaintain::Check
+    metadata do
+      label :puppet_capsules
+      for_feature :foreman_database
+      description 'Check for Puppet capsules from the database'
+      manual_detection
+    end
+
+    def run
+      puppet_proxies = find_puppet_proxies.reject { |proxy| proxy['url'].include?(hostname) }
+      unless puppet_proxies.empty?
+        names = puppet_proxies.map { |proxy| proxy['name'] }
+        question = "You have proxies with Puppet feature enabled\n"\
+                    "The following proxies have Puppet feature: #{names.join(', ')}.\n"\
+                    'Continue Puppet removal?'
+        answer = ask_decision(question, actions_msg: 'y(yes), q(quit)')
+        abort! if answer != :yes
+      end
+    end
+
+    def find_puppet_proxies
+      feature(:foreman_database).query(self.class.puppet_proxies_query)
+    end
+
+    def self.puppet_proxies_query
+      <<-SQL
+        SELECT \"smart_proxies\".*
```

Comment:
I wonder why these need to be escaped.

Reply:
I've copied it over from http_proxies and it works so I didn't question it :shrug: